expr/func_isnonnull.go

```diff
@@ -0,0 +1,62 @@
+package expr
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/grafana/metrictank/api/models"
+	schema "gopkg.in/raintank/schema.v1"
+)
+
+type FuncIsNonNull struct {
+	in GraphiteFunc
+}
+
+func NewIsNonNull() GraphiteFunc {
+	return &FuncIsNonNull{}
+}
+
+func (s *FuncIsNonNull) Signature() ([]Arg, []Arg) {
+	return []Arg{
+		ArgSeriesList{val: &s.in}}, []Arg{ArgSeriesList{}}
+}
+
+func (s *FuncIsNonNull) Context(context Context) Context {
+	return context
+}
+
+func (s *FuncIsNonNull) Exec(cache map[Req][]models.Series) ([]models.Series, error) {
+	series, err := s.in.Exec(cache)
+	if err != nil {
+		return nil, err
+	}
+
+	out := make([]models.Series, 0, len(series))
+	for _, serie := range series {
+		transformed := models.Series{
```

Comment:
the `out` slice is already allocated with the correct size. wouldn't it be better to just update its properties instead of instantiating a new` models.Series` and then copying it into `out`?
f.e.:
`out[i].Target = fmt.Sprintf("isNonNull(%s)", serie.Target),` (`i` is the index of the loop)